src/main/java/datawave/query/util/MetadataHelper.java

```diff
@@ -1070,32 +1062,82 @@ public long getCardinalityForField(String fieldName, Date begin, Date end) throw
     public long getCardinalityForField(String fieldName, String datatype, Date begin, Date end) throws TableNotFoundException {
         log.trace("getCardinalityForField from table: " + metadataTableName);
         Text row = new Text(fieldName.toUpperCase());
+        FrequencyFamilyCounter dateFreqMap = new FrequencyFamilyCounter();
         
         // Get all the rows in DatawaveMetadata for the field, only in the 'f'
-        // colfam
+        // colfamily
         Scanner bs = ScannerHelper.createScanner(connector, metadataTableName, auths);
         
         Key startKey = new Key(row);
         bs.setRange(new Range(startKey, startKey.followingKey(PartialKey.ROW)));
         bs.fetchColumnFamily(ColumnFamilyConstants.COLF_F);
         
         long count = 0;
+        Value aggregatedValue = null;
+        
+        String cq = "";
         
         for (Entry<Key,Value> entry : bs) {
-            Text colq = entry.getKey().getColumnQualifier();
+            if (entry.getKey().getColumnQualifier().toString().startsWith(COL_QUAL_PREFIX))
+                aggregatedValue = entry.getValue();
+            cq = entry.getKey().getColumnQualifier().toString().replaceAll(MetadataHelper.COL_QUAL_PREFIX, "");
+        }
+        
+        if (aggregatedValue == null) {
+            for (Entry<Key,Value> entry : bs) {
```

Comment:
This for loop would never work as we have already exhausted all of the entries in the for loop above.

Reply:
I think this whole section of code can be removed if we are to assume aggregated metadata.

Reply:
Fixed the issue of of exhausting all entries in the loop.  Thanks for catching that.